cabra/src/pages/components/RelationListEdit.tsx

```diff
@@ -0,0 +1,153 @@
+      );
+    },
+  });
+
+  const removeTask = useMutation(
+    (relatedTaskId: string) => deleteTask(relatedTaskId),
+    {
+      onSuccess: () => {
+        client.invalidateQueries(getTasks.cacheKey);
+        client.invalidateQueries(getTask.cacheKey(parentId));
+        client.invalidateQueries(
+          getRelatedTasks.cacheKey(parentId, relationType)
+        );
+      },
+    }
+  );
+
+  const removeRelation = useMutation(
+    (relationId: string) => deleteRelation(relationId),
+    {
+      onSuccess: () => {
+        client.invalidateQueries(getTasks.cacheKey);
+        client.invalidateQueries(getTask.cacheKey(parentId));
+        client.invalidateQueries(
+          getRelatedTasks.cacheKey(parentId, relationType)
+        );
+      },
+    }
+  );
+
+  const handleAddClick = function () {
+    let relationString: string;
+    switch (relationType) {
+      case DirectedRelationType.DependsOn:
+      case DirectedRelationType.IsDependentOn:
+        relationString = "DEPENDENT";
+        break;
+      case DirectedRelationType.Interchangable:
+        relationString = "INTERCHANGABLE";
+        break;
+      default:
+        relationString = "SUBTASK";
+        break;
+    }
+
+    relatedTasks.forEach((t) => {
+      let relation: Omit<ITaskRelation, "id">;
+      if (relationType == DirectedRelationType.IsDependentOn) {
+        relation = {
+          firstTaskId: t.id,
+          secondTaskId: parentId,
+          type: relationString,
+        };
+      } else {
+        relation = {
+          firstTaskId: parentId,
+          secondTaskId: t.id,
+          type: relationString,
+        };
+      }
+      addRelation.mutateAsync(relation);
+    });
+    parentCallback();
+  };
+
+  const handleRemoveClick = function (relationId: string) {
+    removeRelation.mutateAsync(relationId);
+  };
+
+  const { data, isLoading, error } = useQuery(
+    getRelatedTasks.cacheKey(parentId, relationType),
+    () => getRelatedTasks.run(parentId, relationType)
+  );
+
+  if (isLoading) return <Container>Loading</Container>;
+  if (error) return <Container>Oops</Container>;
+  if (!data?.data) return <Container />;
+
+  const allRelatedTasks = data.data.slice().reverse();
+
+  return (
+    <div>
+      <AddSubtaskButton onClick={handleAddClick} tw="m-auto">
+        <PlusIcon width={24} height={24} /> {"Save & Submit Relation"}
+      </AddSubtaskButton>
+      <CenteredLabel>
+        {relationType == DirectedRelationType.DependsOn && "Depends on:"}
+        {relationType == DirectedRelationType.IsDependentOn &&
+          "Is dependent on:"}
+        {relationType == DirectedRelationType.Subtask && "Subtask:"}
+        {relationType == DirectedRelationType.Interchangable &&
+          "Interchangeable:"}
+      </CenteredLabel>
+      <Container>
+        <div tw="text-center">
+          {allRelatedTasks.length == 0 && "<No related task>"}
+        </div>
+        {allRelatedTasks.map((relatedTask) => (
```

Comment:
You can use object destructuring. Ref: https://github.com/airbnb/javascript#destructuring